geoh5py/shared/utils.py

```diff
@@ -16,11 +16,28 @@
 #  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
 
 from abc import ABC
-from typing import List, Optional
+from typing import List, Optional, Union
 
+import h5py
 import numpy as np
 
 
+def fetch_h5_handle(
+    file: Union[str, h5py.File],
+) -> h5py.File:
+    """
+    Open in read+ mode a geoh5 file from string.
+
+    :param file: Name or handle to a geoh5 file.
+
+    :return h5py.File: Handle to an opened h5py file.
+    """
+    if isinstance(file, h5py.File):
+        return file
+
+    return h5py.File(file, "r+")
```

Comment:
see https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager to implement context and be able to use: `with fetch_h5_handle(file) as h5file:`
with no need to worry about closing the file or not

```suggestion
@contextmanager
def fetch_h5_handle(
    file: Union[str, h5py.File],
) -> h5py.File:
    """
    Open in read+ mode a geoh5 file from string.
    If receiving a file instead of a string, merely return the given file.

    :param file: Name or handle to a geoh5 file.

    :return h5py.File: Handle to an opened h5py file.
    """
    if isinstance(file, h5py.File):
        try:
            return file
        finally:
            pass
    else:
        h5file = h5py.File(file, "r+")
        try:
            yield h5file
        finally:
            h5file.close()
```